src/php/Compiler/Domain/Evaluator/Exceptions/CompiledCodeIsMalformedException.php

```diff
@@ -4,13 +4,48 @@
 
 namespace Phel\Compiler\Domain\Evaluator\Exceptions;
 
+use Phel\Compiler\Domain\Analyzer\Ast\AbstractNode;
+use Phel\Compiler\Domain\Analyzer\Ast\Fnable;
+use Phel\Compiler\Domain\Analyzer\Ast\LocalVarNode;
+use Phel\Lang\SourceLocation;
 use RuntimeException;
 use Throwable;
 
 final class CompiledCodeIsMalformedException extends RuntimeException
 {
-    public static function fromThrowable(Throwable $e): self
+    public static function fromThrowable(Throwable $e, AbstractNode $node): self
     {
-        return new self($e->getMessage(), 0, $e);
+        $msg = $e->getMessage();
+
+        if ($node instanceof Fnable) {
+            $msg = self::normalize($e->getMessage(), $node);
+        }
+
+        return new self($msg, 0, $e);
+    }
+
+    private static function normalize(string $msg, Fnable $node): string
+    {
+        $srcLoc = $node->getStartSourceLocation();
+
+        $pattern = '/Too few arguments to function.*, (?<passed>\d+) passed in .* and exactly (?<expected>\d+) expected/';
```

Comment:
The `CompiledCodeIsMalformedException::normalize` method converts the message that starts with `Too few arguments to function` into a user-friendly message.

First, which class throws an exception with a message that starts with `Too few arguments to function`?

I searched but couldn't find it.

Reply:
Look at https://github.com/phel-lang/phel-lang/blob/main/src%2Fphp%2FCompiler%2FDomain%2FEvaluator%2FRequireEvaluator.php lime 37, while doing a require pho code, if that code is malformed, then PHP throws an exception and it's being captured and transformed into our custom code malformed exception

Reply:
*PHP* throw. I understood.
thank you